lib/backend/buffer.go

```diff
@@ -31,11 +32,33 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type bufferConfig struct {
+	gracePeriod time.Duration
+	capacity    int
+}
+
+type BufferOption func(*bufferConfig)
+
+// BufferCapacity sets the event capacity of the circular buffer.
+func BufferCapacity(c int) BufferOption {
+	return func(cfg *bufferConfig) {
+		cfg.capacity = c
```

Comment:
Should we guard against zero or negative capacity? Maybe logging and setting the default?